File: AutoSchedulerBE/AutoScheduler.Domain/Entities/Timesheets/Timesheet.cs
using AutoScheduler.Domain.Entities.Activities;
using AutoScheduler.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AutoScheduler.Domain.Entities.Timesheets
{
    public class Timesheet
    {
        public int Id { get; set; }
        public required string Title { get; set; }
        public TimesheetState State { get; set; }
        public bool Optimized { get; set; }
        public int BaseSlotDuration { get; set; }
        public int BreakDuration { get; set; }
        public IList<Timeslot>? Timeslots { get; set; }
        public IList<ActivityRequirements>? Requirements { get; set; }
        public TimeOnly StartTime { get; set; }
        public TimeOnly EndTime { get; set; }
        public TimeOnly? GeneralBreakStart { get; set; }
        public TimeOnly? GeneralBreakEnd { get; set; }
        public IList<TimesheetActivityRequirements>? TimesheetActivityRequirements { get; set; }

    }
}

File: AutoSchedulerFE/src/classes/timesheet.ts
import type { Activity, ActivityRequirements, Hall } from "./activity";
import type { Group, Member } from "./group";

export enum TimesheetState{
    Draft,
    Active,
    Inactive
}

export interface Timesheet{
    id: number,
    title: string,
    state: TimesheetState,
    optimized: boolean,
    baseSlotDuration: number,
    breakSlotDuration: number,
    timeslots: Timeslot[],
    requirements: ActivityRequirements[],
    startTime: string,
    endTime: string,
    generalBreakStart: string,
    generalBreakEnd: string,
};

export interface Timeslot{
    id: number,
    timesheetId: number,
    activity: Activity,
    hall: Hall,
    member: Member | undefined,
    group: Group,
    dayOfWeek: number,
    startTime: string,
    endTime: string,
    optimizationStatus: string
};

export interface GeneratorRequirements{
    requirements: ActivityRequirements[],
    slotDurationInMinutes: number,
    breakDurationInMinutes: number,
    startTime: string,
    endTime: string,
};

export interface TimeslotPlacementChange{
    generatorRequirements: GeneratorRequirements,
    timeslotsForSheet: Timeslot[] | undefined | null,
    changedTimeslot: Timeslot
};

export interface TimeslotRearrangement{
    generatorRequirements: GeneratorRequirements,
    lockedTimeslots: Timeslot[]
};

export interface WeekdayTimeRange{
    dayOfWeek: number,
    startTime: string,
    endTime: string,
}

export interface TimeslotWeekdayTimeRanges{
    timeslot: Timeslot,
    weekdayTimeRanges:WeekdayTimeRange[]
}

export interface TimesheetViewRequirements{
    slotDurationInMinutes: number,
    breakDurationInMinutes: number,
    startTime: string,
    endTime: string
}